org.eclipse.lsp4e.test/src/org/eclipse/lsp4e/test/AllCleanRule.java

```diff
@@ -53,14 +59,17 @@ protected void finished(Description description) {
 	}
 
 	private void clear() {
-		PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage().closeAllEditors(false);
+		// Give the platform three attempts to shut down windows
+		for (int i = 3; i > 0 && !PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage().closeAllEditors(false); i--) {}
```

Comment:
What's the point of doing this in a loop without a sleep or something in between?

Reply:
It's a fairly long-running call so I didn't feel the need to stick a sleep in between, but I take your point!